x-pack/packages/ml/trained_models_utils/src/constants/trained_models.ts

```diff
@@ -87,16 +93,18 @@ export const ELASTIC_MODEL_DEFINITIONS: Record<string, ModelDefinition> = Object
 } as const);
 
 export interface ModelDefinition {
+  modelName: string;
   version: number;
   config: object;
   description: string;
   os?: string;
   arch?: string;
   default?: boolean;
   recommended?: boolean;
+  hidden?: boolean;
 }
 
-export type ModelDefinitionResponse = ModelDefinition & {
+export type ModelDefinitionResponse = Omit<ModelDefinition, 'modelName'> & {
```

Comment:
In use are `name` and `modelName` the same values?
If so, can this be refactored so we don't need both types?

Reply:
these are different, `name` refers to the actual model id, e.g. `.elser_model_2` and `modelName` is just a helper to determine a model type, e.g. `elser`. `modelName` isn't exposed to the public, hence omitted.